### PR TITLE
Kwarg injection with `@activity` decorated functions

### PR DESCRIPTION
**Pull request by vachillo**
- [x] I have read and agree to the contributing guidelines for [submitting new pull requests](https://github.com/griptape-ai/griptape?tab=readme-ov-file#submitting-pull-requests).

## Describe your changes
Allow arbitrary kwargs in activity-decorated functions

## Issue ticket number and link

----
📚 Documentation preview 📚: https://griptape--1265.org.readthedocs.build//1265/

### CODE REVIEW

File: griptape/utils/decorators.py [hunk 1]
```diff
@@ -24,8 +25,28 @@ def activity(config: dict) -> Any:
...
+            if "params" in func_params or kwarg_var is not None:
+                kwargs["params"] = params
+            if "values" in func_params or kwarg_var is not None:
+                kwargs["values"] = params.get("values")
+
+            for param_name in func_params.keys():  # noqa: SIM118
+                if param_name not in kwargs:
+                    kwargs[param_name] = None
+
+            return func(self, **kwargs)
```
Comment (collindutter) [inline griptape/utils/decorators.py:49]: Can you please move this out to a function and add a test?

File: griptape/utils/decorators.py [hunk 2]
```diff
@@ -24,8 +25,28 @@ def activity(config: dict) -> Any:
...
+                    break
+
+            kwargs = {k: v for k, v in params.get("values", {}).items() if k in func_params or kwarg_var is not None}
+
+            if "params" in func_params or kwarg_var is not None:
+                kwargs["params"] = params
+            if "values" in func_params or kwarg_var is not None:
+                kwargs["values"] = params.get("values")
+
+            for param_name in func_params.keys():  # noqa: SIM118
```
Comment (collindutter) [inline griptape/utils/decorators.py:45]: Why the `noqa`? Seems like a valid lint.

File: griptape/utils/decorators.py [hunk 3]
```diff
@@ -14,7 +15,7 @@
 )
 
 
-def activity(config: dict) -> Any:
+def activity(config: dict) -> Any:  # noqa: C901
```
Comment (collindutter) [inline griptape/utils/decorators.py:18]: Can't ignore your way out of complexity

File: griptape/utils/decorators.py [hunk 4]
```diff
@@ -24,8 +25,28 @@ def activity(config: dict) -> Any:
...
+                if param.kind == inspect.Parameter.VAR_KEYWORD:
+                    kwarg_var = func_params.pop(param.name).name
+                    break
+
+            kwargs = {k: v for k, v in params.get("values", {}).items() if k in func_params or kwarg_var is not None}
+
+            if "params" in func_params or kwarg_var is not None:
+                kwargs["params"] = params
+            if "values" in func_params or kwarg_var is not None:
+                kwargs["values"] = params.get("values")
```
Comment (collindutter) [inline griptape/utils/decorators.py:43]: Can you please add some comments? Lot to unpack here.

**Review by collindutter — CHANGES_REQUESTED** [review]
Awesome

File: griptape/utils/decorators.py (hunk 1 above)
Comment (vachillo) [reply griptape/utils/decorators.py:49]: cov-bot says im good tho

Comment (vachillo) [reply griptape/utils/decorators.py]: but really we dont have decorators.py tests. i added a few methods to `test_base_tool`

Comment (collindutter) [reply griptape/utils/decorators.py]: > cov-bot says im good tho

That guy will approve anything

**Review by collindutter — APPROVED** [review]
Very nice 👍👍

**Pull request merged**